pkg/compactor/bucket_compactor.go

```diff
@@ -797,9 +798,12 @@ func (c *BucketCompactor) Compact(ctx context.Context, maxCompactionTime time.Du
 		// Record the difference between now and the max time for a block being compacted. This
 		// is used to detect compactors not being able to keep up with the rate of blocks being
 		// created. The idea is that most blocks should be for within 24h or 48h.
-		now := time.Now()
-		for _, delta := range c.blockMaxTimeDeltas(now, jobs) {
-			c.metrics.blocksMaxTimeDelta.Observe(delta)
+		{
+			now := time.Now()
+			maxDeltas, ooo := c.blockMaxTimeDeltas(now, jobs)
+			for _, delta := range maxDeltas {
+				c.metrics.blocksMaxTimeDelta.WithLabelValues(strconv.FormatBool(ooo)).Observe(delta)
```

Comment:
What if instead we track each delta based whether it's a OOO block or not? Basically the idea is that `blockMaxTimeDeltas()` returns two slices of `[]float64`: one of the standard blocks and one for the OOO blocks.